hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImplTest.java

```diff
@@ -500,4 +501,28 @@ public void testWaitInitialMembership() {
                 member("127.0.0.3")), UUID.randomUUID());
         clusterService.waitInitialMemberListFetched();
     }
+
+    @Test
+    public void testGetEffectiveMemberList() {
+        ClientClusterServiceImpl clusterService = new ClientClusterServiceImpl(mock(ILogger.class));
+
+        // Returns an empty list on startup, till the members view event
+        assertCollection(Collections.emptyList(), clusterService.getEffectiveMemberList());
+
+        MemberInfo member = member("127.0.0.1");
+        List<MemberInfo> members = Collections.singletonList(member);
+        clusterService.handleMembersViewEvent(1, members, UUID.randomUUID());
+
+        // Returns the member list after the members view event
+        assertCollection(
+                members.stream()
+                        .map(MemberInfo::toMember)
+                        .collect(Collectors.toList()),
+                clusterService.getEffectiveMemberList());
+
+        clusterService.onTryToConnectNextCluster();
+
+        // Returns an empty list after reset
+        assertCollection(Collections.emptyList(), clusterService.getEffectiveMemberList());
```

Comment:
what about adding another handleMembersViewEvent call here and then reassert?

Reply:
I think it won't add any extra value, as it will be the same state changes with the `initial -> event`, but let me put these into a for loop so we do the assertion couple of times.

Reply:
I think it may have slight value to assert that getEffectiveMemberList closely follows memberlist snapshot version and nothing else.